### DISCUSSION TITLE
How to change auto complete color from white to other?

### DISCUSSION THREAD

**Discussion by benyaminl** [chrisant996/clink#315] [General]
https://github.com/chrisant996/clink/issues/314

I already tried 
```
color.doskey
color.selected_completion
color.selection
```

It still shown as white.

**Comment by benyaminl** [comment]
@chrisant996 I think I find the problem... the white alias and black alias is swapped... so I need manually type using sgr.. and the one I should aim is color.cmd

```
color.cmd = sgr 00;34
```

**Reply by chrisant996** [reply]
The alias isn't swapped... The color theme you're using has swapped the color definitions for Black and White inn order to simulate a "light theme".

It has the same effect, but the reason is very different, and the options for fixing it are very different. Clink can't do anything about it.

**Reply by benyaminl** [reply]
yeah yeah (I mean my theme cause it), Because my theme, it cause confusion.

Anyway regarding color.arg should be the color of `/?` from `dir` right? or it should be color.arginfo? 
I tried both with sgr still stuck with bright yellow

**Reply by chrisant996** [reply]
> color.arg should be the color of `/?` from `dir` right?

Not exactly. It's for arguments defined by argmatchers.

> or it should be color.arginfo? 

That's for an arg to a flag, like a filename following some `--file:` flag.

> I tried both with sgr still stuck with bright yellow

I think you're talking about input that isn't categorized as anything in particular. So that's `color.input` if I recall correctly.

Clink isn't coloring based on whether the first character is minus or slash. It colors based on argmatchers. If there's no argmatcher for the command, then it's just generic input.

Btw I have a script I'm working on that can auto-parse help text from programs, in a few different layout formats, to auto-define argmatchers.

**Reply by benyaminl** [reply]
Ugh wow so `color.input`, I see... well anyway thank you. 

And I really thankful for you to bring this project to us. I really really thankful for it. 

Sorry if I done something bad at the issue, and thank you for being nice while replying even I done something dumb.

**Reply by chrisant996** [reply]
> Ugh wow so `color.input`, I see... well anyway thank you. 
> 
> And I really thankful for you to bring this project to us. I really really thankful for it. 
> 
> Sorry if I done something bad at the issue, and thank you for being nice while replying even I done something dumb. 

All good. If you have a moment, maybe you could share some feedback about where/how the documentation could be improved to make the coloring system more clear.

Thanks!

**Reply by chrisant996** [reply]
I added some clarification text to the [Coloring the Input Text](https://chrisant996.github.io/clink/clink.html#coloring-the-input-text) section of the documentation (also reachable from [Colors](https://chrisant996.github.io/clink/clink.html#gettingstarted_colors) in the Getting Started section of the documentation).
